contracts/libraries/ClusterLib.sol

```diff
@@ -80,4 +84,71 @@ library ClusterLib {
                 )
             );
     }
+
+    function validateClusterOnRegistration(
+        ISSVNetworkCore.Cluster memory cluster,
+        uint64[] memory operatorIds,
+        StorageData storage s
+    ) internal view returns (bytes32 hashedCluster) {
+        hashedCluster = keccak256(abi.encodePacked(msg.sender, operatorIds));
+
+        bytes32 clusterData = s.clusters[hashedCluster];
+        if (clusterData == bytes32(0)) {
+            if (
+                cluster.validatorCount != 0 ||
+                cluster.networkFeeIndex != 0 ||
+                cluster.index != 0 ||
+                cluster.balance != 0 ||
+                !cluster.active
+            ) {
+                revert ISSVNetworkCore.IncorrectClusterState();
+            }
+        } else if (clusterData != hashClusterData(cluster)) {
+            revert ISSVNetworkCore.IncorrectClusterState();
+        } else {
+            validateClusterIsNotLiquidated(cluster);
+        }
+    }
+
+    function updateClusterOnRegistration(
+        ISSVNetworkCore.Cluster memory cluster,
+        uint64[] memory operatorIds,
+        bytes32 hashedCluster,
+        uint32 validatorCountDelta,
+        StorageData storage s,
+        StorageProtocol storage sp
+    ) internal {
+        uint64 burnRate;
+
+        if (cluster.active) {
```

Comment:
In the current implementation, the additional check for the **cluster.active** flag in our function may be redundant, as the `cluster.validateClusterOnRegistration` function already includes this check internally. 
To enhance code efficiency and clarity, I suggest removing the `cluster.active `check from our function.